### PR TITLE
Bump crucible to get the Const_RefRoots fix.

### LINKED ISSUE #2064: MIR override incorrectly rejected when applied to multiple `const` slice arguments
Consider this program:

```rs
// test.rs
const X: &[u32] = &[0; 4];
const Y: &[u32] = &[1; 4];

pub fn f(x: &[u32], y: &[u32]) -> bool {
    x[0] == y[0]
}

pub fn g() -> bool {
    f(X, Y)
}
```

I would expect SAW to be able to verify `f` and `g` against `f_spec` and `g_spec` below:

```
// test.saw
enable_experimental;

let f_spec = do {
  x_ref <- mir_alloc (mir_array 4 mir_u32);
  x_val <- mir_fresh_var "x" (mir_array 4 mir_u32);
  mir_points_to x_ref (mir_term x_val);

  y_ref <- mir_alloc (mir_array 4 mir_u32);
  y_val <- mir_fresh_var "y" (mir_array 4 mir_u32);
  mir_points_to y_ref (mir_term y_val);

  mir_execute_func [mir_slice_value x_ref, mir_slice_value y_ref];

  mir_return (mir_term {{ x_val@0 == y_val@0 }});
};

let g_spec = do {
  mir_execute_func [];

  mir_return (mir_term {{ False }});
};

m <- mir_load_module "test.linked-mir.json";

f_ov <- mir_verify m "test::f" [] false f_spec z3;
mir_verify m "test::g" [f_ov] false g_spec z3;
```

Surprisingly, however, SAW fails when using the `f_ov` override in a compositional proof of `g_spec`:

```
$ saw-rust test.rs
$ cabal run exe:saw -- test.saw
[17:07:05.306] Loading file "/home/ryanscott/Documents/Hacking/Haskell/sandbox/saw-script/test.saw"
[17:07:05.346] Verifying test/c3adb09d::f[0] ...
[17:07:05.360] Simulating test/c3adb09d::f[0] ...
[17:07:05.361] Checking proof obligations test/c3adb09d::f[0] ...
[17:07:05.377] Proof succeeded! test/c3adb09d::f[0]
[17:07:05.402] Verifying test/c3adb09d::g[0] ...
[17:07:05.411] Simulating test/c3adb09d::g[0] ...
[17:07:05.411] Matching 1 overrides of  test/c3adb09d::f[0] ...
[17:07:05.411] Stack trace:
"mir_verify" (/home/ryanscott/Documents/Hacking/Haskell/sandbox/saw-script/test.saw:27:1-27:11)
Symbolic execution failed.
Abort due to assertion failure:
  test.rs:10:10: 10:11: error: in test/c3adb09d::g[0]
  attempted to read empty mux tree
```

This is very strange. What's more, `crux-mir` is able to simulate the same program (using `g` as a `#[crux::test]` entrypoint), which suggests that the fault lies with SAW. Moreover, SAW _is_ able to handle the following, simpler program without issues:

```rs
// works.rs
const X: &[u32] = &[0; 4];

pub fn f(x: &[u32]) -> u32 {
    x[0]
}

pub fn g() -> u32 {
    f(X)
}
```
```
// works.saw
enable_experimental;

let f_spec = do {
  x_ref <- mir_alloc (mir_array 4 mir_u32);
  x_val <- mir_fresh_var "x" (mir_array 4 mir_u32);
  mir_points_to x_ref (mir_term x_val);

  mir_execute_func [mir_slice_value x_ref];

  mir_return (mir_term {{ x_val@0 }});
};

let g_spec = do {
  mir_execute_func [];

  mir_return (mir_term {{ 0 : [32] }});
};

m <- mir_load_module "wat.linked-mir.json";

f_ov <- mir_verify m "wat::f" [] false f_spec z3;
mir_verify m "wat::g" [f_ov] false g_spec z3;
```

This suggests that the problem only arises when an override has multiple references as arguments. This led me to discover that if you comment out this line of code:

https://github.com/GaloisInc/saw-script/blob/45d64dd7e56c2879cd0a684be321d579899e3404/src/SAWScript/Crucible/MIR/Override.hs#L947

Then SAW will accept the specs in `test.saw` that it previously rejected. I'm not exactly sure what is going on, but this suggests that something is rather broken in the way that disjointness checking is currently implemented.

### PR DESCRIPTION
**Pull request by sauclovian-g**
Includes the test from the issue.

Fixes #2064.

### CODE REVIEW

**Comment by RyanGlScott** [comment]
Looking good!

Note that the MIR-related test cases in this repo have checked in their MIR JSON files to version control so that the test suite can be ran without needing to install `mir-json` beforehand. The downside to this approach is that these JSON files are still using the old `mir-json` schema (prior to the changes in https://github.com/GaloisInc/mir-json/pull/73 and https://github.com/GaloisInc/crucible/pull/1243), so they will no longer parse with the new version of `crucible-mir`. This can be seen in:

1. The `saw` integration tests (failing CI log [here](https://github.com/GaloisInc/saw-script/actions/runs/10568338784/job/29280163295?pr=2107))
2. The `saw-remote-api` tests (failing CI log [here](https://github.com/GaloisInc/saw-script/actions/runs/10568338784/job/29280164540?pr=2107))

(I'm a bit undecided about whether we want to check in all of the MIR JSON files to version control going forward, but that's the way it's currently done.)

In both cases, the most direct solution would be to generate the MIR JSON files. They are located in the following places:

1. In the `saw` integration tests: All of the directories that are shown when running the `find intTests/ -iname "*.linked-mir.json"` command
2. In the `saw-remote-api` tests: In [this directory](https://github.com/GaloisInc/saw-script/tree/fca29c36de9861fa2811735088e863bf37200d34/saw-remote-api/python/tests/saw/test-files)

In each location, there should be a `Makefile` that can be used to regenerate the JSON files. Please let me know if you have any difficulties.

**Comment by sauclovian-g** [comment]
Bleah, I should have thought of that. After all, I noticed the json files were checked in...

(btw there's a third set in the rust tutorial)

It occurs to me, also, that being able to diff them after a change like this might be worth the space overhead of including newlines in the output...

**Review by RyanGlScott — APPROVED** [review]

**Comment by RyanGlScott** [comment]
Ah, one last thing: you'll have to update [these lines](https://github.com/GaloisInc/saw-script/blob/fca29c36de9861fa2811735088e863bf37200d34/intTests/test_mir_verify_basic/test.saw#L75-L77) to reflect the new disambiguator values:

```diff
 // Test using fully disambiguated names
-mir_verify m "test/775505e0::id_u8" [] false (id_spec mir_u8) z3;
-mir_verify m "test/775505e0::id_u8[0]" [] false (id_spec mir_u8) z3;
+mir_verify m "test/ff7472fa::id_u8" [] false (id_spec mir_u8) z3;
+mir_verify m "test/ff7472fa::id_u8[0]" [] false (id_spec mir_u8) z3;
```

**Comment by sauclovian-g** [comment]
I have done so. I wouldn't have thought this would have changed much of anything here, though. So I have investigated, which may or may not have been worthwhile; the change from 775505e0 to ff7472fa covers the whole compilation and isn't particularly interesting.

What's different though is that the original has two extra copies of ```id_array```, specialized to take no args and return the array size (one each for the argument type and return type sizes, even though they're the same); these have disappeared. Is this because of a different compiler version (or just because it's Olde), which probably doesn't matter, or reflect something about compiler options or modes, which may?

**Comment by RyanGlScott** [comment]
**tl;dr** These changes are benign.

> What's different though is that the original has two extra copies of `id_array`, specialized to take no args and return the array size (one each for the argument type and return type sizes, even though they're the same); these have disappeared. 

In more concrete terms: before this PR, `id_array` was compiled to the following MIR:

```rs
fn test/775505e0::id_array[0](_1 : [u32; 5]) -> [u32; 5] {
   let mut _0 : [u32; 5];
   bb0: {
      _0 = use(_1);
      return;
   }
}
fn test/775505e0::id_array[0]::{constant#0}[0]() -> usize {
   let mut _0 : usize;
   bb0: {
      _0 = use(5);
      return;
   }
}
fn test/775505e0::id_array[0]::{constant#1}[0]() -> usize {
   let mut _0 : usize;
   bb0: {
      _0 = use(5);
      return;
   }
}
```

I believe `id_array[0]::{constant#0}` and `id_array[0]::{constant#1}` represent the lengths of the first and second array types, respectively, in `id_array`'s type signature. In this small program, they are not referenced anywhere else in the generated MIR, although it is conceivable that one could cook up a more complicated program where they are needed.

After this PR, however, neither `id_array[0]::{constant#0}` and `id_array[0]::{constant#1}` appear at all in the generated MIR. This is expected behavior, and these changes are benign. These changes aren't due to https://github.com/GaloisInc/mir-json/pull/73, but rather due to a different commit (https://github.com/GaloisInc/mir-json/pull/56) that was brought in by recompiling this program with a more recent version of `mir-json`. (That commit didn't change the JSON schema in any way, so we didn't bother to recompile SAW's test cases at the time.) 

As for why that commit changes the MIR JSON output: https://github.com/GaloisInc/mir-json/pull/56 causes `saw-rustc` only to mark functions as roots when performing dead-code elimination. Since `id_array[0]::{constant#0}` and `id_array[0]::{constant#1}` are considered top-level constants rather than functions at the MIR level, they are no longer marked as roots. Moreover, since no other definition references them, they are now discarded during `mir-json`'s dead-code elimination pass.

**Comment by sauclovian-g** [comment]
Good to know, thanks. So I think we can go ahead here...
